### ISSUE TITLE
Label autocompletion seems wrong when using local labels

### ISSUE THREAD

**Issue by rgiot** [mborik/z80-macroasm-vscode#2]
Let's use this example:

```z80
global
	ld hl, global
	ld de,  ; Cursor is before the comment

.local
```

If I type `.l<tab>` I expect to obtain `.local`.
Instead I obtain `.global.local` that is totally invalid.
 `global.local` would be valid, but it is preferable to not have the `global` part as we are already in its scope.

**Comment by mborik** [comment]
Nesting of labels is behavior known from SjASMPlus, see [this doc](http://z00m128.github.io/sjasmplus/documentation.html#idp17).
Have you any suggestion how I detect when you want to type a "local" label and when not.
Extension building a table of symbols from a start of file (including includes) and all locals are appended after the global symbol.
I don't know how you want to solve it. It is still just a suggestion, you know 😉

**Comment by rgiot** [comment]
I have probably badly explained. The behavior I expect is 100% compatible with sjasmplus one.
To clarify, here is the obtained code after completion

```z80
global
	ld hl, global
	ld de,  .global.local

.local
```

Wheras the sjasmplus (or rasm or vasm) code would be

```z80
global
	ld hl, global
	ld de,  global.local

.local
```

There is an extra `.` that prefix the autocompleted label and that should not be there.

Are we talking of 2 different things ?

```z80
global
	ld hl, global
	ld de,  .local

.local
```
is also valid because the `ld de, .local` line is in the `global` context.

**Comment by rgiot** [comment]
By the way, I have noticed another illustration of this issue:

```z80
global
.local<enter>
```

is transformed as

```z80
global
.global.local
```

instead of being

```z80
global
.local
```


But if I do:

```z80
global
.local


global2
.local2<enter>
```

this issue is not raised

**Comment by mborik** [comment]
Ok, I see it now. I need to think about it and find proper solution for that kind of "proximity label suggestion". :)

**Comment by maziac** [comment]
Same problem here. Is there any solution to this meanwhile?

**Comment by mborik** [comment]
solved by 6b0f70c
